### PR TITLE
Allow marvin to listen to channel messages

### PR DESCRIPTION
**Pull request by Arkham**
:point_up: :dancer:

### CODE REVIEW

File: lib/marvin/core.ex [hunk 1]
```diff
@@ -23,12 +23,15 @@ defmodule Marvin.Core do
...
-        scrubbed_message = message |> scrub_indentifier(slack)
-        dispatch_message(:direct, scrubbed_message, slack)
+      cond do
+        String.match?(message.text, ~r/#{slack.me.id}/) ->
+          scrubbed_message = message |> scrub_indentifier(slack)
+          dispatch_message(:direct, scrubbed_message, slack)
+        String.match?(message.channel, ~r/^D/) ->
+          dispatch_message(:direct, message, slack)
+        true ->
+          dispatch_message(:message, message, slack)
```
Comment (willrax) [inline lib/marvin/core.ex]: Do you think :ambient would work as the matcher for this?

Comment (Arkham) [inline lib/marvin/core.ex]: Unfortunately the type of the message is literally 'message' :D

Comment (Arkham) [inline lib/marvin/core.ex]: Also why are we doing this part?

```
       String.match?(message.channel, ~r/^D/) ->
         dispatch_message(:direct, message, slack)
```

Comment (willrax) [inline lib/marvin/core.ex]: We can call them what ever we want though

Comment (willrax) [inline lib/marvin/core.ex]: that bit matches against a direct message with the bot

Comment (Arkham) [inline lib/marvin/core.ex]: done!

File: lib/marvin/core.ex [hunk 2]
```diff
@@ -55,6 +58,13 @@ defmodule Marvin.Core do
     end)
   end
 
+  defp dispatch_message(:ambient, message, slack) do
+    Application.get_env(:marvin, :bots)
+    |> Enum.each(fn(bot) ->
+      if bot.is_match?({:message, message.text}), do: bot.handle_event(message, slack)
```
Comment (willrax) [inline lib/marvin/core.ex]: this will have to be :ambient as well in order to match

Comment (willrax) [inline lib/marvin/core.ex]: then in your bot you would use match {:ambient, ~r/spotify/}

**Comment by willrax** [comment]
:+1:

**Pull request merged**